opentracing/src/main/java/io/opentracing/Tracer.java

```diff
@@ -87,15 +87,18 @@
       /** Same as {@link Span#setTag(String, String)}, but for the span being built. */
       SpanBuilder withTag(String key, String value);
 
-      /** Same as {@link Span#setTag(String, String)}, but for the span being built. */
+      /** Same as {@link Span#setTag(String, boolean)}, but for the span being built. */
       SpanBuilder withTag(String key, boolean value);
 
-      /** Same as {@link Span#setTag(String, String)}, but for the span being built. */
+      /** Same as {@link Span#setTag(String, Number)}, but for the span being built. */
       SpanBuilder withTag(String key, Number value);
 
       /** Specify a timestamp of when the Span was started, represented in microseconds since epoch. */
       SpanBuilder withStartTimestamp(long microseconds);
 
+      /** Same as {@link Span#setBaggageItem(String, String)}, but for the span being built. */
+      SpanBuilder withBaggage(String key, String value);
```

Comment:
I thought the whole point of baggage was to be able to propagate state across RPC's.  So my assumption was baggage needs to be read from the RPC handler on the receiving service and added to the new span being created, so that on the next RPC baggage can continue to be passed along.